flink-runtime/src/main/java/org/apache/flink/runtime/state/heap/HeapKeyedStateBackend.java

```diff
@@ -475,6 +482,32 @@ public LocalRecoveryConfig getLocalRecoveryConfig() {
         return localRecoveryConfig;
     }
 
+    @Override
+    public void setup(@Nonnull StateRequestHandler stateRequestHandler) {}
+
+    @Nonnull
+    @Override
+    public <N, S extends org.apache.flink.api.common.state.v2.State, SV> S createState(
+            @Nonnull N defaultNamespace,
+            @Nonnull TypeSerializer<N> namespaceSerializer,
+            @Nonnull org.apache.flink.runtime.state.v2.StateDescriptor<SV> stateDesc)
+            throws Exception {
+        StateDescriptorTransformer transformer = new StateDescriptorTransformer();
+        StateDescriptor stateDescV1 = transformer.getStateDescriptor(stateDesc);
+        State state = getOrCreateKeyedState(namespaceSerializer, stateDescV1);
+        if (stateDescV1.getType() == StateDescriptor.Type.VALUE) {
+            return (S) new ValueStateWrapper((ValueState) state);
+        }
+        throw new UnsupportedOperationException(
+                String.format("Unsupported state type: %s", stateDesc.getType()));
+    }
+
+    @Nonnull
+    @Override
+    public StateExecutor createStateExecutor() {
+        return null;
+    }
+
```

Comment:
I suggest move this part to a wrapper/adaptor converting a `KeyedStateBackend` to `AsyncKeyedStateBackend`. The adaptor should exist in `v2.adaptor` package. And we could wrap `KeyedStateBackend` instance in `StreamTaskStateInitializerImpl#streamOperatorStateContext`, WDYT?